### DISCUSSION TITLE
I tried to make my library into an executable using pyinstaller, but pyinstaller can't seem to find my library.

### DISCUSSION THREAD

**Discussion by 3QNRpDwD** [pyinstaller/pyinstaller#7388] [Help]
First of all, please understand that I am not good at English, so I use Google Translate..
To study the https protocol, I implemented the https protocol with Python, and uploaded my library to the pypi site to test if communication with the outside was possible and used pyinstaller to create an executable file for the client and server. However, pyinstaller could not find my library. printed an error message. If anyone knows a solution, please let me know

![Desktop Screenshot 2023 01 22 - 01 43 10 01](https://user-images.githubusercontent.com/120706785/213877424-ed662cb7-20d5-4c4f-ba62-03ee898162f1.png)
![Desktop Screenshot 2023 01 22 - 01 43 17 13](https://user-images.githubusercontent.com/120706785/213877427-c3fa57fc-a0b8-4766-8ca6-5757521556d8.png)

**Comment by rokm** [comment]
The server is failing because you are trying to list the contents of the package's directory:
https://github.com/projectlonginus/LonginusPYPI/blob/c342c7bbe665b291650d29e3d01296a6c5de82a7/package/LServer.py#L120

In PyInstaller-frozen applications, the python modules are collected in executable-embedded PYZ archive, so there are no on-disk package directories, unless the package also contains non-python files (e.g., C extension modules or data files). If  `Sessions`, `Session_data`, `Session_keys`, `port`, `addr', and `Login_list`  are supposed to be static data files, you should collect them into your frozen application (which will also ensure that package directory exists). 

If they are not static (e.g., are generated at run-time), then you should find a better place to store them, because the package directory might in read-only location (this applies to un-frozen python code as well, not only PyInstaller-frozen applications, which are supposed to be immutable). Store them in a cache directory in the user's home directory, e.g. using [`appdirs`](https://pypi.org/project/appdirs/) to determine correct location for your program.

---

The client is failing because you are trying to use `input()` in a frozen application that was created in the `noconsole`/`windowed` mode, and therefore does not have a console.

**Reply by 3QNRpDwD** [reply]
Thank you very much! it was a big help
